core/domain/suggestion_registry.py

```diff
@@ -635,6 +635,209 @@ def pre_update_validate(self, change):
+        """Validates the BaseVoiceoverApplication object.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                BaseVoiceoverApplication object are invalid.
+        """
+
+        if self.target_type not in suggestion_models.TARGET_TYPE_CHOICES:
+            raise utils.ValidationError(
+                'Expected target_type to be among allowed choices, '
+                'received %s' % self.target_type)
+
+        if not isinstance(self.target_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected target_id to be a string, received %s' % type(
+                    self.target_id))
+
+        if self.status not in suggestion_models.STATUS_CHOICES:
+            raise utils.ValidationError(
+                'Expected status to be among allowed choices, '
+                'received %s' % self.status)
+
+        if not isinstance(self.author_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected author_id to be a string, received %s' % type(
+                    self.author_id))
+        if self.status == suggestion_models.STATUS_IN_REVIEW:
+            if self.final_reviewer_id is not None:
+                raise utils.ValidationError(
+                    'Expected final_reviewer_id to be None as the '
+                    'voiceover application is not yet handled.')
+        else:
+            if not isinstance(self.final_reviewer_id, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected final_reviewer_id to be a string, received %s' % (
+                        type(self.final_reviewer_id)))
+            if self.status == suggestion_models.STATUS_REJECTED:
+                if not isinstance(
+                        self.rejection_message, python_utils.BASESTRING):
+                    raise utils.ValidationError(
+                        'Expected rejection_message to be a string for a '
+                        'rejected application, received %s' % type(
+                            self.final_reviewer_id))
+            if self.status == suggestion_models.STATUS_ACCEPTED:
+                if self.rejection_message is not None:
+                    raise utils.ValidationError(
+                        'Expected rejection_message to be None for the '
+                        'accepted voiceover application, received %s' % (
+                            self.rejection_message))
+
+        if not isinstance(self.language_code, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected language_code to be a string, received %s' %
+                self.language_code)
+        if not utils.is_supported_audio_language_code(self.language_code):
+            raise utils.ValidationError(
+                'Invalid language_code: %s' % self.language_code)
+
+        if not isinstance(self.filename, python_utils.BASESTRING):
```

Comment:
Maybe also validate that the filetype is a supported audio filetype?

Reply:
That's a part of audio validation which will be validated based on the audio type we handel and it will be part of audio service which stores/validates/decodes audio. 

Basically this domain layer expects to have an audio filename in string with any format and the audio services will handel the type of the audio (this will help us to extend our audio services seperately)